runtimes/data-binding.mdx

```diff
@@ -0,0 +1,169 @@
+    | Boolean           | False           |
+    | Color             | #00000000       |
+    | Trigger           | Untriggered     |
+    | Enum              | The first value |
+    | Nested view model | Null            |
+
+2. Create the default instance - Use the instance labelled "Default" in the editor. Usually this is the one a designer intends as the primary one to be used at runtime.
+
+3. Create by index - Using the order returned when iterating over all available instances. Useful when creating multiple instances by iteration.
+
+4. Create by name - Use the editor's instance name. Useful when creating a specific instance.
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        val vm = view.controller.file?.getViewModelByName("My VM")!!
+
+        // Create blank
+        val vmiBlank = vm.createBlankInstance()
+
+        // Create default
+        val vmiDefault = vm.createDefaultInstance()
+
+        // Create by index
+        for (i in 0 until vm.instanceCount) {
+            val vmiIndexed = vm.createInstanceFromIndex(i)
+        }
+
+        // Create by name
+        val vmiNamed = vm.createInstanceFromName("My Instance")
+        ```
+    </Tab>
+</Tabs>
+
+# View Model Instances
+
+Once you've used your view model to create an instance, that instance can be assigned to an artboard. This establishes the bindings set up at edit time.
+
+<Note>The initial values of the instance are not applied to their bound elements until the artboard is advanced.</Note>
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        view.setRiveResource(
+            R.raw.my_rive_file,
+            artboardName = "My Artboard",
+        )
+
+        val vm = view.controller.file?.getViewModelByName("My View Model")!!
+        val vmi = vm.createInstanceFromName("My Instance")
+
+        // Apply the instance to the artboard
+        view.controller.activeArtboard?.setViewModel(vmi)
+        ```
+    </Tab>
+</Tabs>
+
+# Properties
+
+Properties descriptors can be inspected on a view model to discover at runtime which are available. These are not the mutable properties themselves though - once again those are on instances. These descriptors have a type and name.
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        val vm = view.controller.file?.getViewModelByName("My View Model")!!
+
+        // A list of properties
+        val properties = vm.properties
+        assertContains(
+            properties,
+            ViewModel.Property(ViewModel.PropertyDataType.NUMBER, "My Number Property")
+        )
+        ```
+    </Tab>
+</Tabs>
+
+View model instances have mutable properties. References to these properties can be retrieved by name. They have get, set, and observe operations. Getting or observing the value will retrieve the latest value set on that properties binding, as of the last artboard advance. Setting the value will update the value and all of its bound elements. Note that trigger properties can only be set.
```

Comment:
"state machine advance"

Reply:
You can also register callbacks on `trigger` properties (not get, but just observe when it's triggered_. You might need to double-check that Android implemented that. I also missed that behavior in my initial implementation for rive_native.